decoder/src/lib.rs

```diff
@@ -588,6 +588,159 @@ mod tests {
         );
     }
 
+    #[test]
+    fn display_use_inner_type_hint() {
+        let mut entries = BTreeMap::new();
+
+        entries.insert(
+            0,
+            TableEntry::new_without_symbol(Tag::Info, "x={:b}".to_owned()),
+        );
+        entries.insert(
+            1,
+            TableEntry::new_without_symbol(Tag::Derived, "S {{ x: {=u8:x} }}".to_owned()),
+        );
+
+        let table = Table {
+            entries,
+            timestamp: Some(TableEntry::new_without_symbol(
+                Tag::Timestamp,
+                "{=u8:µs}".to_owned(),
+            )),
+        };
+
+        let bytes = [
+            0,  // index
+            2,  // timestamp
+            1,  // index of the struct
+            42, // value
+        ];
+
+        let frame = table.decode(&bytes).unwrap().0;
+        assert_eq!(
+            frame.display(false).to_string(),
+            "0.000002 INFO x=S { x: 0x2a }",
+        );
+    }
+
+    #[test]
+    fn display_use_outer_type_hint() {
+        let mut entries = BTreeMap::new();
+
+        entries.insert(
+            0,
+            TableEntry::new_without_symbol(Tag::Info, "x={:b}".to_owned()),
+        );
+        entries.insert(
+            1,
+            TableEntry::new_without_symbol(Tag::Derived, "S {{ x: {=u8:?} }}".to_owned()),
+        );
+
+        let table = Table {
+            entries,
```

Comment:
What do you think about this pattern? It would also be applicable for some other tests.
```suggestion
        let entries = vec![
            TableEntry::new_without_symbol(Tag::Info, "x={:b}".to_owned()),
            TableEntry::new_without_symbol(Tag::Derived, "S {{ x: {=u8:x} }}".to_owned()),
        ];

        let table = Table {
            entries: entries.into_iter().enumerate().collect::<BTreeMap<_, _>>(),
```